earthkit/geo/distance.py

```diff
@@ -139,45 +146,59 @@ def nearest_point_haversine(ref_points, points):
     res_index = []
     res_distance = []
     for lat, lon in ref_points.T:
-        distance = haversine_distance((lat, lon), points).flatten()
+        distance = haversine_distance((lat, lon), points, radius=1).flatten()
         index = np.nanargmin(distance)
         index = index[0] if isinstance(index, np.ndarray) else index
         res_index.append(index)
         res_distance.append(distance[index])
-    return (np.array(res_index), np.array(res_distance))
+    return (np.array(res_index), np.array(res_distance) * radius)
 
 
-def ll_to_xyz(lat, lon):
+def _ll_to_xyz(lat, lon, radius=constants.R_earth):
     lat = np.asarray(lat)
     lon = np.asarray(lon)
     lat = np.radians(lat)
     lon = np.radians(lon)
-    x = constants.R_earth * np.cos(lat) * np.cos(lon)
-    y = constants.R_earth * np.cos(lat) * np.sin(lon)
-    z = constants.R_earth * np.sin(lat)
-    return x, y, z
+    x = np.cos(lat) * np.cos(lon)
+    y = np.cos(lat) * np.sin(lon)
+    z = np.sin(lat)
+
+    if radius == 1:
+        return x, y, z
+    else:
+        return radius * x, radius * y, radius * z
 
 
-def cordlength_to_arclength(chord_length):
+def _cordlength_to_arclength(chord_length, radius=constants.R_earth):
     """
     Convert 3D (Euclidean) distance to great circle arc length
     https://en.wikipedia.org/wiki/Great-circle_distance
     """
-    central_angle = 2.0 * np.arcsin(chord_length / (2.0 * constants.R_earth))
-    return constants.R_earth * central_angle
+    if radius == 1:
```

Comment:
I think we should not do it this way; we should instead have a supporting object (a "figure", which means a "shape" and a "size" -- and we want to control both), which could be Sphere(R) (R as argument), UnitSphere (Sphere1?), SphereIFS, OblateSpheroid(a, b), OblateSpheroid(R, e) (eccentricity), and so on with methods for all we need. This, together with utility functions that would take radius or a,b as arguments. Also SphereIFS could be Earth, or Earth = Sphere(6666666)

If we are concerned with speed (I don't think we should really) we should also have a variant latr,lonr (in radian instead of degree.)

Reply:
Thanks. I implemented the `figure` option.